### PR TITLE
[Impl] Write Decision data structure of OVM #19

### PR DESCRIPTION
**Pull request by syuhei176**
### Task List
- Create ovm folder
  - Add challenges to Decision
  - Add Not operator
  - Add Boolean decider
- Apply Coder.Address and Coder.Bytes

### CODE REVIEW

File: src/ovm/interfaces/IDeciderManager.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Decider } from './Decider'
+import { Address } from '../../types'
+import { Property, Decision } from '../types'
+
+export interface IDeciderManager {
```
Comment (syuhei176) [inline src/ovm/interfaces/IDeciderManager.ts]: @tkmct Do you think we require IDeciderManager? and what do you think about name?

Comment (tkmct) [reply src/ovm/interfaces/IDeciderManager.ts]: I don't think we need the interface.
DeciderManager only needs concrete implementation as a singleton.

Comment (tkmct) [reply src/ovm/interfaces/IDeciderManager.ts]: But if we need custom DeciderManager for different main chain, I prefer extending the DeciderManager class.

Comment (syuhei176) [reply src/ovm/interfaces/IDeciderManager.ts]: maybe, we don't need custom DeciderManager for each L1.

**Comment by tkmct** [comment]
Please migrate with new data primitives!!

**Comment by syuhei176** [comment]
maybe passed

**Comment by syuhei176** [comment]
I will split PR

**Comment by syuhei176** [comment]
@tkmct It became big and it has 2 changes.
- Adding OVM folder with NotDecider
- Applying Coder.Address and Coder.Bytes to all modules
can you review?
I tried to divide it, but it was a little messy.

File: __tests__/ovm/NotDecider.test.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    )
+    expect(decision.outcome).toEqual(true)
+  })
+  it('decide not(true)', async () => {
+    const deciderManager = new DeciderManager()
+    deciderManager.setDecider(SampleDeciderAddress, new SampleDecider())
+    deciderManager.setDecider(NotDeciderAddress, new NotDecider())
+    const decision = await deciderManager.decide(
+      new Property(NotDeciderAddress, [
+        Bytes.from(utils.arrayify(EthCoder.encode(trueProperty.toStruct())))
```
Comment (syuhei176) [inline __tests__/ovm/NotDecider.test.ts]: I suppose If `EthCoder.encode()` returns 'Coder.Bytes', it becomes more clean.

Comment (tkmct) [reply __tests__/ovm/NotDecider.test.ts]: Great idea!!

**Comment by tkmct** [comment]
I'll take a look!

File: src/types/Codables/Bytes.ts [hunk 3]
```diff
@@ -15,6 +15,13 @@ export default class Bytes implements Codable {
     return new Bytes(u)
   }
 
+  static fromHexString(hex: string): Bytes {
```
Comment (tkmct) [inline src/types/Codables/Bytes.ts]: Thank you! This makes a lot of sense.

**Pull request merged**